prems-resources/backend/src/main/java/io/uhndata/cards/prems/internal/importer/AbstractConditionalClarityDataProcessor.java

```diff
@@ -0,0 +1,182 @@
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import io.uhndata.cards.clarity.importer.spi.ClarityDataProcessor;
+
+/**
+ * Clarity import processor that checks if all provided conditions are matched.
+ *
+ * @version $Id$
+ */
+public abstract class AbstractConditionalClarityDataProcessor implements ClarityDataProcessor
+{
+    protected final int priority;
+
+    protected final List<ConditionDefinition> conditions;
+
+    AbstractConditionalClarityDataProcessor(int priority, String[] conditionStrings)
+    {
+        this.priority = priority;
+        this.conditions = new ArrayList<>(conditionStrings.length);
+        for (String conditionString : conditionStrings) {
+            ConditionDefinition def = new ConditionDefinition(conditionString);
+            if (def.isValid()) {
+                this.conditions.add(def);
+            }
+        }
+    }
+
+    protected abstract Map<String, String> handleUnmatchedCondition(Map<String, String> input);
+
+    protected abstract Map<String, String> handleAllConditionsMatched(Map<String, String> input);
+
+    @Override
+    public Map<String, String> processEntry(Map<String, String> input)
+    {
+        for (ConditionDefinition condition : this.conditions) {
+            if (!condition.matches(input)) {
+                return this.handleUnmatchedCondition(input);
+            }
+        }
+        // All filters match if data got here
+        return this.handleAllConditionsMatched(input);
+    }
+
+    @Override
+    public int getPriority()
+    {
+        return this.priority;
+    }
+
+    private enum Operator
+    {
+        NEQ("<>", false, (input, value) -> input == null || !input.equalsIgnoreCase(value)),
+        LTE("<=", false, (input, value) -> compareDouble(input, value, (a, b) -> a <= b)),
+        LT("<", false, (input, value) -> compareDouble(input, value, (a, b) -> a < b)),
+        GTE(">=", false, (input, value) -> compareDouble(input, value, (a, b) -> a >= b)),
+        GT(">", false, (input, value) -> compareDouble(input, value, (a, b) -> a > b)),
+        EQ("=", false, (input, value) -> input != null && input.equalsIgnoreCase(value)),
+        NOT_MATCHES("not matches", false, (input, value) -> input == null || !input.matches(value)),
+        MATCHES("matches", false, (input, value) -> input != null && input.matches(value)),
+        NOT_IN("not in", false, (input, value) -> input == null
+            && !Arrays.stream(value.split("\\s*;\\s*")).anyMatch(input::equalsIgnoreCase)),
+        IN("in", false, (input, value) -> input != null
+            && Arrays.stream(value.split("\\s*;\\s*")).anyMatch(input::equalsIgnoreCase)),
+        EMPTY("is empty", true, (input, value) -> input == null || input.length() == 0),
+        NOT_EMPTY("is not empty", true, (input, value) -> input != null && input.length() > 0);
```

Comment:
```suggestion
        NOT_MATCHES(" not matches ", false, (input, value) -> input == null || !input.matches(value)),
        MATCHES(" matches ", false, (input, value) -> input != null && input.matches(value)),
        NOT_IN(" not in ", false, (input, value) -> input == null
            && !Arrays.stream(value.split("\\s*+;\\s*+")).anyMatch(input::equalsIgnoreCase)),
        IN(" in ", false, (input, value) -> input != null
            && Arrays.stream(value.split("\\s*+;\\s*+")).anyMatch(input::equalsIgnoreCase)),
        EMPTY(" is empty", true, (input, value) -> input == null || input.length() == 0),
        NOT_EMPTY(" is not empty", true, (input, value) -> input != null && input.length() > 0);
```

`*+` is a [possessive matcher](https://www.regular-expressions.info/possessive.html) that improves performance by grabbing as many spaces as possible and not backtracking into them when failing to match the rest of the expression.

Reply:
It would improve performance if the regexp is compiled.

Reply:
Regex not implemented, strings and possesive matcher implemented